src/cuda/api/multi_wrapper_impls/kernel_launch.hpp

```diff
@@ -29,8 +30,9 @@ void enqueue_launch(
 	KernelParameters&&...   parameters)
 {
 	static_assert(
-		detail_::all_true<::std::is_trivially_copy_constructible<detail_::kernel_parameter_decay_t<KernelParameters>>::value...>::value,
-		"All kernel parameter types must be of a trivially copy-constructible (decayed) type." );
+		detail_::all_true<traits::is_valid_kernel_argument<detail_::kernel_parameter_decay_t<KernelParameters>>::value...>::value,
```

Comment:
The new comment is a bit vague, and even a bit misleading: All you need in order to be a kernel parameter is to be blessed with a specialization of traits. But you could instead refer to the trait's documentation, if that one has the actual requirements.